packages/client/src/definitions/LinkDefinitions.ts

```diff
@@ -41,24 +38,34 @@ export type OsdkObjectLinksEntry<
   O extends ObjectTypeDefinition<any>,
   L extends ObjectTypeLinkKeysFrom2<O>,
 > = O["links"][L] extends ObjectTypeLinkDefinition<infer T, infer M> ? (
-    M extends false ? SingletonLinkAccessor<T> : MultitonLinkAccessor<T>
+    M extends false ? SingleLinkAccessor<T> : MultiLinkAccessor<T>
   )
   : never;
 
-export interface SingletonLinkAccessor<T extends ObjectTypeDefinition<any>> {
+export type SingleLinkReturnType<
+  T extends ObjectTypeDefinition<any>,
+  A extends SelectArg<T>,
+> = Promise<Osdk<T, SelectArgToKeys<T, A>>>;
+
+export type MultiLinkReturnType<
+  T extends ObjectTypeDefinition<any>,
+  A extends SelectArg<T>,
+> = Promise<PageResult<Osdk<T, SelectArgToKeys<T, A>>>>;
+
+export interface SingleLinkAccessor<T extends ObjectTypeDefinition<any>> {
   /** Load the linked object */
-  get: <A extends SelectArg<T>>(
+  get: <const A extends SelectArg<T>>(
     options?: A,
-  ) => Promise<OsdkObjectFrom<T, SelectArgToKeys<A>>>;
+  ) => SingleLinkReturnType<T, A>;
 }
 
-export interface MultitonLinkAccessor<T extends ObjectTypeDefinition<any>> {
-  get: <A extends SelectArg<T>>(
+export interface MultiLinkAccessor<T extends ObjectTypeDefinition<any>> {
```

Comment:
We wanted to switch this over to just being a new object set creator. Do you want to tackle this now or another PR? We should probably do it in a single 'breaking' change for upstream at least.

Reply:
This change is massive already so lets save it.